x-pack/plugins/infra/public/pages/metrics/hosts_landing/index.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import { EuiButton, EuiCallOut } from '@elastic/eui';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { EuiLink } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { enableInfrastructureHostsView } from '@kbn/observability-plugin/public';
+import { HostsPage } from '../hosts';
+import { EnableHostViewPage } from './enable_host_view_page';
+
+export const HostsLandingPage = () => {
+  const kibana = useKibana();
+  const canEditAdvancedSettings = kibana.services.application?.capabilities.advancedSettings.save;
+  const [isHostViewEnabled, setIsHostViewEnabled] = useState<boolean | undefined>(
```

Comment:
nit: you could also use this observable `uiSettings.get$` to capture changes in `enableInfrastructureHostsView`, without having to store a state in the component:

```ts
const isHostViewEnabled = useObservable<boolean>(
    uiSettings?.get$<boolean>(enableInfrastructureHostsView) ??
      new Observable((subs) => subs.next(false))
  );
```

Unfortunately we need this `new Observable((subs) => subs.next(false)` because `uiSettings` can be undefined.

Reply:
That's a nice suggestion, thank you, I didn't know about this observable. Works fine, I changed that part 🙂